lib/semantics.py

```diff
@@ -0,0 +1,62 @@
+import gensim.downloader as api
+from gensim.models import KeyedVectors
+from gensim.models.fasttext import load_facebook_model
+import os
+import os.path as path
+import sys
+import urllib.request as request
+import zipfile
+
+
+class SimilarWordMap:
+    """
+    Defines the base-level requirements for a SimilarWordMap. Must provide:
+      1) Top k most similar words
+      2) Similarity metric between two words
+    """
+
+    def get_similar_words(self, word, count):
+        """
+        Returns the top k (where  k = count) most similar words to the provided word.
+        """
+        raise NotImplementedError()
+
+
+class FastTextSimilarWordMap(SimilarWordMap):
+    DOWNLOAD_LINK = "https://dl.fbaipublicfiles.com/fasttext/vectors-english/wiki-news-300d-1M.vec.zip"
```

Comment:
could we all store this locally for now so that we don't have to download this every time we run it?

Reply:
Already does! 😄It just automatically downloads it for you if you don't have it downloaded yet.

https://github.com/lbialik/dadbot/pull/6/files#diff-070e56c815647ecad6af04944c480af4R37-R44